### PR TITLE
various minor fixes

### PR DESCRIPTION
**Pull request by ebakker1**
A record of some changes I made to get the training script running: given my machine is less powerful than a uni cluster I was having some memory problems without them.

I've added comments to explain the changes. This branch is a bit behind the actual PR branch, so not sure it's worth merging, but more for discussion.

### CODE REVIEW

File: MoleculeVAE-reImplementation/dataloader.py [hunk 1]
```diff
@@ -195,7 +198,7 @@ def caculate_target_values(smiles):
     dataset.
     """
     m = Chem.MolFromSmiles(smiles)
-    logp = Chem.Descriptors.MolLogP(m)
```
Comment (ebakker1) [inline MoleculeVAE-reImplementation/dataloader.py:198]: Calling this way didn't work for my version of `RDkit` but maybe that's my environment.

File: MoleculeVAE-reImplementation/train.py [hunk 2]
```diff
@@ -167,30 +169,33 @@ def load_data_by_task(params):
...
-        X_train, X_test, character_index_lookup_dict = load_dataset(params)
+        X_train, X_test, _ = load_dataset(params)
 
     train_dataset = CustomMoleculeDataset(X_train, Y_train)
-    train_set_final, valid_set_final = split_validation_dataset(train_dataset, percentage_train=0.8)
+    train_dataset, valid_dataset = split_validation_dataset(
+        train_dataset, percentage_train=0.8)
     test_dataset = CustomMoleculeDataset(X_test, Y_test)
-    test_set_final, _ = split_validation_dataset(test_dataset, percentage_train=1)# To control the number of percentage
-    training_data_dic = pd.DataFrame(list(zip( train_set_final, valid_set_final, test_dataset )),
```
Comment (ebakker1) [inline MoleculeVAE-reImplementation/train.py:178]: This seemed to be a bit confused. The variable is called `..._dic` but you're making it into a `DataFrame`, and the items going into the `DataFrame` are 2 `torch.DataLoader` objects and one `torch.DataSet` object. This caused my VM to crash, presumably because it wasn't sure what to do with that output.
Later `DataFrame` like actions (like `to_list` get called on this output but in the final `train` and `test` calls they seem to be being treated as `DataLoader` or `DataSet` types. 

I tried to straighten it all out by making `training_data_dic` and dictionary holding `DataSet` objects and did the conversion to `DataLoaders` inside the `train` and `test` functions.

File: MoleculeVAE-reImplementation/train.py [hunk 3]
```diff
@@ -167,30 +169,33 @@ def load_data_by_task(params):
...
+        train_set_final=train_dataset,
+        valid_set_final=valid_dataset,
+        test_dataset=test_dataset)
+    return training_data_dict
 
 
 
 '''Loss function '''
 
-def vae_loss(x_decoded_mean, x, z_mean, z_logvar):
```
Comment (ebakker1) [inline MoleculeVAE-reImplementation/train.py:186]: `vae_loss` is here used as a function name but in other places is used as a variable name. This leads my IDE to get confused and possibly also the code (though I can't verify that). 
Confusions like this are part of why it's good to keep variables as nouns and functions as verbs. This is also what's recommended by PEP.

File: MoleculeVAE-reImplementation/train.py [hunk 4]
```diff
@@ -211,20 +216,20 @@ def calculate_loss_by_type(decoder_output_data, input_string_data, z_mean, z_log
...
-        loss = vae_loss(decoder_output_data, input_string_data, z_mean, z_logvar)
+        loss = calc_vae_loss(decoder_output_data, input_string_data, z_mean, z_logvar)
     elif loss_type == "pro_prediction_only":
         loss = pre_loss
 
     elif loss_type == "vae_pre_no_annealing":
-        vecloss,kl_loss = vae_loss(decoder_output_data, input_string_data, z_mean, z_logvar)
+        vecloss,kl_loss = calc_vae_loss(decoder_output_data, input_string_data, z_mean, z_logvar)
         #TODO:scaling factor
         vae_loss=vecloss+kl_loss
```
Comment (ebakker1) [inline MoleculeVAE-reImplementation/train.py:227]: Example of use as a variable name, conflicting with function definition.

File: MoleculeVAE-reImplementation/train.py [hunk 5]
```diff
@@ -307,13 +322,18 @@ def train(epochs,train_set_final, valid_set_final,logger):
...
 
                     print(f'{epochs} / {batch_idx}\t{loss:.4f}')
-        return (train_loss.cpu().detach().numpy() / len(train_set_final)), (valid_loss.cpu().detach().numpy() / len(valid_set_final))
-
-def test(test_set,epochs,logger):
-    dataset=torch.utils.data.DataLoader(test_set, batch_size=params['batch_size'])
+        return (train_loss.cpu().detach().numpy() / len(train_dataloader)), (valid_loss.cpu().detach().numpy() / len(valid_dataloader))
+
+def test(
+        test_set: data.Dataset,
```
Comment (ebakker1) [inline MoleculeVAE-reImplementation/train.py:346]: I've typed the inputs here to try and avoid confusions about what should be passed in. This helps a good IDE flag is something unexpected is going into the function.

File: MoleculeVAE-reImplementation/train.py [hunk 6]
```diff
@@ -307,13 +322,18 @@ def train(epochs,train_set_final, valid_set_final,logger):
...
+        epochs,
+        logger,
+        params: Dict,
+        device):
```
Comment (ebakker1) [inline MoleculeVAE-reImplementation/train.py:350]: Since we are now doing the `DataLoader` conversion in this function we need the `params` here to get the `batch_size`.
I've also passed in `device`. It's not clear to me where `device` was coming from in the previous version of the code since it wasn't getting passed in or defined within the namespace.

File: MoleculeVAE-reImplementation/train.py [hunk 7]
```diff
@@ -279,57 +285,102 @@ def plot3D(xi, yi, zi, c, label, title, n_min, n_max):
...
             optimizer.zero_grad()
             output, mean, logvar,pre,z = model(data)
             pre_loss_ca=nn.MSELoss()
             pre_loss=pre_loss_ca(pre,label)
-            loss=calculate_loss_by_type(output,data, mean, logvar,pre_loss,epochs,'train',logger)
+            loss=calculate_loss_by_type(
+                output,data, mean, logvar,pre_loss,epoch,'train',logger)
             loss.backward()
-            train_loss += loss
+            train_loss += loss.cpu().detach()
```
Comment (ebakker1) [inline MoleculeVAE-reImplementation/train.py:318]: Without this `detach` and a similar one in the validation loop my RAM was getting swamped.

File: MoleculeVAE-reImplementation/train.py [hunk 8]
```diff
@@ -279,57 +285,102 @@ def plot3D(xi, yi, zi, c, label, title, n_min, n_max):
...
             optimizer.step()
-            if batch_idx % len(train_set_final) == 0:
```
Comment (ebakker1) [inline MoleculeVAE-reImplementation/train.py:298]: This seems to just do validation at the beginning of the whole training loop, so I instead removed this and did the validation loss as a separate loop outside the training loop.

File: MoleculeVAE-reImplementation/train.py [hunk 9]
```diff
@@ -279,57 +285,102 @@ def plot3D(xi, yi, zi, c, label, title, n_min, n_max):
...
+                valid_pre_loss_ca = nn.MSELoss()
+                valid_pre_loss = valid_pre_loss_ca(
+                    valid_pre,
+                    valid_labels.to(dtype=torch.float32, device=device))
+                valid_loss += calculate_loss_by_type(
+                    valid_output, valid_smidata,
+                    valid_mean,
+                    valid_logvar,
+                    valid_pre_loss,
+                    epoch,'valid',logger).cpu().detach()
```
Comment (ebakker1) [inline MoleculeVAE-reImplementation/train.py:337]: As with above, had memory problems without this `detach`

File: MoleculeVAE-reImplementation/train.py [hunk 10]
```diff
@@ -279,57 +285,102 @@ def plot3D(xi, yi, zi, c, label, title, n_min, n_max):
 
 '''Train & Test func.'''
 
-def train(epochs,train_set_final, valid_set_final,logger):
+def train(
+        model: torch.nn.Module,
```
Comment (ebakker1) [inline MoleculeVAE-reImplementation/train.py:289]: Changed to passing in `model` as an input. As with `device` this makes the dependency explicit, which I think is clearer.
